docs/internal-documentation/templates/specifications/tests/Template01.md

```diff
@@ -0,0 +1,267 @@
+
+## Objective
+
+> > Write a description of what this test case tests, i.e. the details that are
+> > considered to be right and wrong. Include references to RFCs and other
+> > documents.
+> >
+> > Do not include full links here (in the source). Put them at the end of the
+> > specification document instead, as in this template.
+> >
+> > Do have deep links, but keep the display text short if possible. E.g.:
+
+...described in [RFC 4035][RFC 4035#section-3.1.3], section 3.1.3, ...
+
+## Scope
+
+> > If this test case assumes, but not depends on, another test case then specify
+> > it. Dependency would mean that the test case is affected by the other test
+> > case, or even cannot be run if the other has not been run. Assuming that
+> > another test case has been run is just to make it possible to ignore certain
+> > errors. E.g.:
+
+It is assumed that *Child Zone* has been tested and reported by [Basic04]. This
+test case will just ignore non-responsive name servers or name servers not
+giving a correct DNS response for an authoritative name server.
+
+
+## Inputs
+
+> > Input data to the test case. Always include the Child zone, but it can also
+> > be other data units.
+> >
+> > The input data name is put in quote marks '""', then a hyphen '-' as a
+> > separator, and finally a description.
+
+* "Child Zone" - The domain name to be tested.
+
+## Summary
+
+> > First we can have bullets, if applicable, that state notable things about
+> > the execution or the messages. E.g.
+* If no CDS record is found, the test case will terminate early
+  with no message tag outputted.
+* If a CDS record is of "delete" type, then it can by definition not
+  match or point at any DNSKEY record.
+
+> > Here is a table of all message tags referred to in the steps. The tag in the
+> > first column, the default severity level in the second, and a statement on
+> > when the message is outputted in the fourth.
+> >
+> > If data from the test, e.g. list of name server IP addresses, is to be
+> > outputted with the message, then the datatypes are listed in the third
+> > column, "arguments". The third column is left empty when no arguments are
+> > used for the message.
+> >
+> > Always use the same table set-up, but with the correct tags. E.g.:
+
+Message Tag outputted         | Level   | Arguments  | Description of when message tag is outputted
+:-----------------------------|:--------|:-----------|:--------------------------------------------
+T01_BROKEN_DNSSEC             | ERROR   | ns_ip_list | Replies do not follow the standard.
+T01_ALGO_NOT_SUPPORTED_BY_ZM  | NOTICE  | algo_descr, algo_num | The algorithm used is not supported by the Zonemaster implementation.
+T01_HAS_NSEC                  | INFO    |            | The *Child Zone* uses NSEC.
+T01_HAS_NSEC3                 | INFO    |            | The *Child Zone* uses NSEC3.
+T01_INCONSISTENT_DNSSEC       | ERROR   | keytag     | The configuration of the zone is inconsistent with respect to DNSSEC.
```

Comment:
Should the array be sorted by "Message tag" ?

Reply:
Yes, it should. Good catch. I have updated the table. Please re-approve.